### PR TITLE
refactor: use FormBuilder for account form

### PR DESCRIPTION
**Pull request by thesanjeevsharma**
Related to #2350 

I've made the changes to the Account form. Please let me know if the approach is correct and if the same needs to be done for other forms.

(I am react dev, touched angular after years 🙈 )

### CODE REVIEW

File: apps/client/src/app/pages/accounts/create-or-update-account-dialog/create-or-update-account-dialog.component.ts [hunk 1]
```diff
@@ -26,14 +30,45 @@ export class CreateOrUpdateAccountDialog implements OnDestroy {
   public constructor(
     private dataService: DataService,
     public dialogRef: MatDialogRef<CreateOrUpdateAccountDialog>,
-    @Inject(MAT_DIALOG_DATA) public data: CreateOrUpdateAccountDialogParams
+    @Inject(MAT_DIALOG_DATA) public data: CreateOrUpdateAccountDialogParams,
```
Comment (dtslvr) [inline apps/client/src/app/pages/accounts/create-or-update-account-dialog/create-or-update-account-dialog.component.ts:33]: Can you move up ⬆️ for an alphabetical order?

File: apps/client/src/app/pages/accounts/create-or-update-account-dialog/create-or-update-account-dialog.component.ts [hunk 2]
```diff
@@ -26,14 +30,45 @@ export class CreateOrUpdateAccountDialog implements OnDestroy {
...
     this.platforms = platforms;
+
+    this.accountForm = this.formBuilder.group({
+      accountId: [this.data.account.id],
+      balance: [this.data.account.balance, Validators.required],
+      comment: [this.data.account.comment],
+      currency: [this.data.account.currency, Validators.required],
+      isExcluded: [this.data.account.isExcluded],
+      platformId: [this.data.account.platformId],
+      name: [this.data.account.name, Validators.required]
```
Comment (dtslvr) [inline apps/client/src/app/pages/accounts/create-or-update-account-dialog/create-or-update-account-dialog.component.ts:50]: Can you move up ⬆️ for an alphabetical order?

File: apps/client/src/app/pages/accounts/create-or-update-account-dialog/create-or-update-account-dialog.html [hunk 3]
```diff
@@ -68,9 +70,8 @@ <h1 *ngIf="!data.account.id" i18n mat-dialog-title>Add account</h1>
         <mat-label i18n>Account ID</mat-label>
         <input
           disabled
```
Comment (dtslvr) [inline apps/client/src/app/pages/accounts/create-or-update-account-dialog/create-or-update-account-dialog.html:72]: This does not seem to work anymore. Can you adapt it for form control?

File: apps/api/src/app/account/create-account.dto.ts [hunk 4]
```diff
@@ -12,7 +12,7 @@ import { isString } from 'lodash';
 export class CreateAccountDto {
   @IsOptional()
   @IsString()
-  accountType: AccountType;
+  accountType?: AccountType;
```
Comment (dtslvr) [inline apps/api/src/app/account/create-account.dto.ts:15]: 👍🏻

File: apps/api/src/app/account/update-account.dto.ts [hunk 5]
```diff
@@ -12,7 +12,7 @@ import { isString } from 'lodash';
 export class UpdateAccountDto {
   @IsOptional()
   @IsString()
-  accountType: AccountType;
+  accountType?: AccountType;
```
Comment (dtslvr) [inline apps/api/src/app/account/update-account.dto.ts:15]: 👍🏻

**Review by dtslvr — CHANGES_REQUESTED** [review]
This already looks very good @thesanjeevsharma. I have added a few comments.

**Comment by thesanjeevsharma** [comment]
@dtslvr Thanks for the review. I've addressed the comments.

**Review by dtslvr — APPROVED** [review]
Thank you @thesanjeevsharma for the changes. Looks fine now! I have made some minor improvements.

**Pull request merged**